crates/iceberg/src/transaction/snapshot.rs

```diff
@@ -410,6 +444,7 @@ impl<'a> SnapshotProducer<'a> {
                 self.table.metadata().current_snapshot_id(),
                 next_seq_num,
                 Some(first_row_id),
+                compression,
```

Comment:
```suggestion
                compression.clone(),
```
should this be .clone?  or should the rest be without .clone?